lib/land/trackers/api_tracker.rb

```diff
@@ -206,24 +227,69 @@ def user_agent
                      Land.config.blank_user_agent_string
 
         @user_agent = UserAgent[user_agent]
+        @user_agent.user_agent_type = UserAgentType['user']
 
         if Land.config.identify_crawlers && defined?(CrawlerDetect)
           crawler_detect = CrawlerDetect.new(user_agent)
-          user_agent_type = crawler_detect.is_crawler? ? 'crawl' : 'api'
-          @user_agent.update(user_agent_type: UserAgentType[user_agent_type])
+          @user_agent.user_agent_type = UserAgentType['crawl'] if crawler_detect.is_crawler?
         end
 
         browser = ::Browser.new(user_agent)
 
-        @user_agent.update(
-          browser: Browser[browser.name],
-          device: Device[browser.device.name],
-          platform: Platform[browser.platform.name],
-          browser_version: browser.version
-        )
+        update_browser_color_preferences(Browser[browser.name])
+
+        @user_agent.browser = Browser[browser.name]
+        @user_agent.device = Device[browser.device.name]
+        @user_agent.platform = Platform[browser.platform.name]
+        @user_agent.browser_version = browser.version
+        @user_agent.device_resolution = device_resolution
+
+        begin
+          @user_agent.save! if @user_agent.changed?
+        rescue Standard::Error => e
+          add_error_tag_to_land_span(e)
+        end
 
         @user_agent
       end
+
+      def device_resolution
+        return unless device_width && device_height
+
+        resolution = DeviceResolution.find_or_initialize_by(
+          width: device_width,
+          height: device_height,
+          orientation: device_orientation
+        )
+
+        resolution.device_resolution = "#{device_width}x#{device_height}"
+
+        resolution.save! if resolution.changed?
+        resolution
+      rescue StandardError => e
+        add_error_tag_to_land_span(e)
+      end
```

Comment:
This will have a race condition possible, this will need to be handled with a retry like other calls in this file to find_or_create_by or find_or_initialize_by

Reply:
Please check the error messages that are production in Lab, specifically all errors that originate from the land code.